### PR TITLE
Handle child process termination

### PR DESCRIPTION
**Pull request by shYkiSto**
Fixes an issue where the `jazelle` process exits before the spawned child processes terminates, which leads to a funky experience as the child may continue to write to the `stdout`, which negatively affects the dev experience making the open terminal hard to continue to work with. Since for the most part the `jazelle` commands are just wrapping other tooling (emulating the environment by running correct binaries), it seems reasonable for it to propagate any termination signals to the child and wait till it exits (the same also applies to the exit status, which should be bubbled up). 

- Keep track of active child processes that are spawned using `spawn()` from the `utils/node-helpers.js`
- Add `SIGINT` signal handler that is no-op, which should prevent the `jazelle` parent process from exiting, with the expectation that `Ctrl+C` sequence on the terminal will send the `SIGINT` signal to all processes for the currently running process group
- Propagate `SIGTERM` to all spawned child processes, so that the underlying program gets a chance to gracefully terminate
- Similarly handle killing the child processes on process exit
- Add `spawnOrExit()` method that can be used in commands that are wrapping other tooling (`bazel`, `yarn`, `node` etc.) to ensure the child process exit code is bubbled up properly (there's an outstanding issue w/ Yarn v2, where it always returns `exitCode = 1` for non-zero exit code: see https://github.com/yarnpkg/berry/pull/4798)
- Make use of async `spawnOrExit()` when executing a script w/ `bazel` to ensure the parent process can handle `SIGTERM` signal propagation

### CODE REVIEW

**Comment by lhorie** [comment]
Is there ever a case where we don't want to spawnAndExit? If not, it might make more sense to just always export spawnAndExit as spawn

**Review by lhorie — APPROVED** [review]

File: rules/execute-command.js [hunk 1]
```diff
@@ -101,13 +105,12 @@ function runCommand(command, args = []) {
     const cmd = command
       .replace(/\$\{NODE\}/g, exe)
       .replace(/\$\{ROOT_DIR\}/g, rootDir);
-    execOrExit(cmd, options);
+    await spawnOrExit(cmd, [], options);
   } else if (command in rootScripts) {
     // if command exists at root level but not at project level, run the root level command instead of erroring
-    execOrExit(`${node} ${yarn} run ${command} ${params}`, {
+    await spawnOrExit(`${node} ${yarn} run ${command} ${params}`, [], {
```
Comment (lhorie) [inline rules/execute-command.js:111]: Can you pass arguments to spawn like this?

Comment (shYkiSto) [reply rules/execute-command.js:111]: Yup, this is possible using `shell: true` option: https://github.com/uber-web/jazelle/blob/22e5853ead0ad8325f7c21c105a60d985688adb6/rules/execute-command.js#L90

It's technically only needs to use shell in this case: https://github.com/uber-web/jazelle/blob/22e5853ead0ad8325f7c21c105a60d985688adb6/rules/execute-command.js#L97-L108, where we take `command` as "granted" (i.e., user input may contain something that requires shell to interpret it). In other cases I could probably change it to use plain args

Comment (lhorie) [reply rules/execute-command.js:111]: 👍

**Review by AlexMSmithCA — APPROVED** [review]

**Review by lhorie — APPROVED** [review]

**Pull request merged**